File: Strategy Pattern/CSharp/Duck/Duck.cs
using System;

using FlyBehavior = Behavior.FlyBehavior;
using QuackBehavior = Behavior.QuackBehavior;

namespace Duck 
{
    public abstract class Duck 
    {
        protected FlyBehavior flyBehavior; 
        protected QuackBehavior quackBehavior;

        public Duck(FlyBehavior fb, QuackBehavior qb) 
        {
            flyBehavior = fb;
            quackBehavior = qb;
        }

        public abstract void display();

        public void swim() => Console.WriteLine("I Can Swim.");
        
        public void performFly() => flyBehavior.fly();
        public void performQuack() => quackBehavior.quack();
    }
}

File: Strategy Pattern/Cpp/Duck/Duck.cpp
#include <iostream>
#include "../Behavior/FlyBehavior.cpp"
#include "../Behavior/QuackBehavior.cpp"

using namespace std;

class Duck {
protected:
	FlyBehavior* flyBehavior;
	QuackBehavior* quackBehavior;

public:
	Duck() { }

	virtual void display() = 0;

	void swim();

	void performFly();
	void performQuack();

	void setFlyBehavior(FlyBehavior* fb);
	void setQuackBehavior(QuackBehavior* qb);
};

void Duck::swim() {
	cout << "I Can Swim.\n";
}

void Duck::performFly() {
	flyBehavior->fly();
}

void Duck::performQuack() {
	quackBehavior->quack();
}

void Duck::setFlyBehavior(FlyBehavior* fb) {
	flyBehavior = fb;
}

void Duck::setQuackBehavior(QuackBehavior* qb) {
	quackBehavior = qb;
}
